code/workspaces/client-api/src/dataaccess/projectService.js

```diff
@@ -102,6 +55,12 @@ const generateOptions = token => ({
   },
 });
 
+export function projectActionRequestToProject(actionRequest) {
```

Comment:
This is interesting. I get why you need it and don't think you should change it in this PR but makes me wonder whether our API in the infra service isn't quite right to support our usage.

Reply:
I thought this too. I didn't know whether doing things through deletion requests etc. were GraphQL specific and it would therefore be wrong to have them go all the way to the infra service. It seemed like it was more of a job for the client api to handle this pattern rather than the infra service.